mesa_geo/geoagent.py

```diff
@@ -103,12 +107,12 @@ def from_GeoDataFrame(self, gdf, unique_id="index", set_attributes=True):
         agents = list()
 
         for index, row in gdf.iterrows():
-            shape = row.geometry
-            new_agent = self.create_agent(shape=shape, unique_id=index)
+            geometry = row[gdf.geometry.name]
+            new_agent = self.create_agent(geometry=geometry, unique_id=index)
 
             if set_attributes:
                 for col in row.index:
-                    if not col == "geometry":
+                    if not col == gdf.geometry.name:
```

Comment:
If this is no longer hardcoded to "geometry", what are the possible values for `gdf.geometry.name`? Is it "geometry" most of the time?

Reply:
This can be quite confusing with GeoPandas:

1.  The default name for spatial feature column is `geometry` for Shapefiles and GeoDataFrame. So if we have the following data:

| | name | geometry |
| :---: | :---: | :---: |
| 0 | Alice | POINT (1.00000 2.00000) |
| 1 | Bob | POINT (2.00000 1.00000) |

then it gets loaded as `agent.name` and `agent.geometry` respectively.

2. But, GeoPandas provides the flexibility to rename the `geometry` column to whatever you want: [Active geometry handling](https://geopandas.org/en/stable/docs/reference/geodataframe.html?highlight=geometry#active-geometry-handling). It'll be useful when you want to have a different name other than the default, or you have multiple spatial feature columns. In any case there'll be only one active `geometry` attribute, and can still be accessed via `gdf.geometry`. For example:

| | name | home | work |
| :---: | :---: | :---: | :---: |
| 0 | Alice | POINT (1.00000 2.00000) | POINT (3.00000 4.00000) |
| 1 | Bob | POINT (2.00000 1.00000) | POINT (4.00000 5.00000) |

If for instance, `home` is the active geometry column, then `gdf.geometry` and `gdf.home` are essentially the same thing. And it'll be loaded as `agent.name`, `agent.geometry`, and `agent.work` respectively.

With previously hard-coded "geometry", it'll be loaded as `agent.name`, `agent.geometry`, `agent.work`, and `agent.home` which duplicates `agent.geometry`.

Just now I updated the test case to verify this.